sphinx/writers/latex.py

```diff
@@ -1123,6 +1127,12 @@ def visit_table(self, node):
                 logger.info(__('both tabularcolumns and :widths: option are given. '
                                ':widths: is ignored.'), location=node)
         self.next_table_colspec = None
+        if self.builder.config.latex_booktabs:
+            self.table.booktabs = True
+            self.table.colsep = ''
+        else:
+            self.table.booktabs = False
+            self.table.colsep = '|'
```

Comment:
Could you initialize these members inside `Table.__init__()` please? I think defining new members outside of the class is not good.